### PR TITLE
Feature000

### PR DESCRIPTION
**Pull request by ffahimi**
quick solution to allow stop losses going through, how can we have a mechanism that sees it's a stop loss? inside marketorderhandler?

### CODE REVIEW

**Comment by ffahimi** [comment]
Added a parameter and a function, to be able to reject orders even if a single leg is in an existing position, similar to your function, just checking each leg against contracts

File: Monitor/Base.py [hunk 1]
```diff
@@ -50,6 +50,12 @@ def parameter(cls, key, default=None):
     # @param algorithm [QCAlgorithm] The algorithm argument that the methods receive is an instance of the base QCAlgorithm class, not your subclass of it.
     # @param targets [List[PortfolioTarget]] The list of targets to be ordered
     def ManageRisk(self, algorithm: QCAlgorithm, targets: List[PortfolioTarget]) -> List[PortfolioTarget]:
+
+        risk_check_times = [time(9, 45, 0), time(10, 15, 0), time(12, 30, 0), time(13, 0, 0), time(13, 30, 0), time(13, 45, 0), time(14, 0, 0), time(15, 0, 0), time(15, 15, 0), time(15, 45, 0)]
+        current_time = self.context.time.time()
+        if current_time not in risk_check_times:
+            return []
```
Comment (Chocksy) [inline Monitor/Base.py:57]: lets not merge these fixed times that seem to be strategy specific in the main code.
We can add an update to the line below that calls `self.preManageRisk()` by making a check for the return and if it returns something like skip or a value we could return with [].

Though i think the best way is to still use managePositionFrequency.

In our chats we did discuss that the frequency does not work over multiple days but we can add an improvement either in this PR or in a separate one to address that.

File: Alpha/Base.py [hunk 2]
```diff
@@ -450,6 +455,40 @@ def buildOrderPosition(self, order, lastClosedOrderTag=None):
 
         return [position, workingOrder]
 
+    def hasOneDuplicateLeg(self, order):
```
Comment (Chocksy) [inline Alpha/Base.py:458]: this method does not seem to get called anywhere now.

**Review by Chocksy — CHANGES_REQUESTED** [review]

File: Monitor/Base.py (hunk 1 above)
Comment (ffahimi) [reply Monitor/Base.py:57]: True, I started using managePositionFrequency, but I guess for more of a control over times we check to close positions, the hours can be used.

The frequency for the stop loss checks rarely is more than a day, so I think minutes is sufficient.

**Pull request closed without merging**